astropy/time/formats.py

```diff
@@ -969,8 +969,16 @@ def parse_string(self, timestr, subfmts):
                 self._scale = scale
                 self._fits_scale = fits_scale
                 self._fits_realization = fits_realization
-            elif (scale != self.scale or fits_scale != self._fits_scale or
-                  fits_realization != self._fits_realization):
+            elif scale == self.scale:
```

Comment:
This is a very nice catch! But I would write the logic slightly differently so that (at least for me) it is clearer what is happening. Specifically, I would do an explicit check for values not set first:
```
elif scale == self.scale and self._fits_scale is None and self._fits_realization is None:
    self._fits_scale = fits_scale
    self._fits_realization = fits_realization
elif scale != self.scale or ... # as it was
    raise ...
```
That way we also do not have to duplicate the error.